lib/codecs/corejson.js

```diff
@@ -1,12 +1,22 @@
 const document = require('../document')
 
+function primitiveToNode (data) {
+  if (data instanceof Object && data._type === 'document') {
```

Comment:
Some reading:

http://stackoverflow.com/questions/899574/which-is-best-to-use-typeof-or-instanceof

http://stackoverflow.com/questions/22289727/difference-between-using-array-isarray-and-instanceof-array